lib/matrix.gi

```diff
@@ -4052,7 +4052,7 @@ InstallGlobalFunction( DirectSumMat, function (arg)
           if IsSubset(c,F) then
             F:=c;
           else
-            Error("build field/ring");
+            F:=Field(Concatenation(GeneratorsOfField(F),GeneratorsOfField(c)));
```

Comment:
I noticed there is an (undocumented, but used) operation `ClosureField`, an alias for `ClosureDivisionRing` (which has a GAPDoc comment, but it is not linked to by the reference manual). I don't think it does any better right now, but in theory a more clever implementation could be added there one day? So perhaps:
```suggestion
            F:=ClosureField(F,c);
```